tests/System.Text.Primitives.Tests/Formatting/TimeTests.cs

```diff
@@ -51,6 +53,7 @@ public void RandomDateTimeTests(char format)
         static void TestDateTimeFormat(DateTime dt, char format, SymbolTable symbolTable)
         {
             var expected = dt.ToString(format.ToString(), CultureInfo.InvariantCulture);
+            if (format == 'r') expected = expected.ToLowerInvariant();
```

Comment:
Headers and url are in practice 7-bit ascii (single byte); so it should always be a non-resizing conversion for date for `x-ms-date` to `Authorization` header. Maybe need an `asciiString` type? 😉